jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultProjectAccountSettingsManager.kt

```diff
@@ -49,7 +53,7 @@ class DefaultProjectAccountSettingsManager(private val project: Project) : Proje
                 CredentialManager.getInstance().getCredentialIdentifierById(credentialId)
             }
 
-            val regionId = state.activeRegion ?: AwsRegionProvider.getInstance().defaultRegion().id
+            val regionId = state.activeRegion ?: credentials?.defaultRegion ?: AwsRegionProvider.getInstance().defaultRegion().id
```

Comment:
defaultRegion looks like it can throw because fallbackRegion can throw: fallbackRegion(): AwsRegion = regions(DEFAULT_PARTITION) which makes it possible to not set a default region?

Reply:
I've rejigged this a bit : https://github.com/aws/aws-toolkit-jetbrains/pull/1760

Basically it'll only be possible to throw if there's no 'us-east-1' in the endpoints data.